### PR TITLE
update doc to reflect latest version, add helm link

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

update doc for external cloud provider ...

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [ ] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: docs/book/src/topics/external-cloud-provider.md [hunk 1]
```diff
@@ -9,20 +9,22 @@
...
     ```
 
-- Deploy a CNI solution
+- Deploy a CNI solution (using Calico now)
+
+    Note: choose desired version by replace <v3.23> below
 
     ```shell
-    curl https://docs.projectcalico.org/v3.19/manifests/calico.yaml | sed "s/veth_mtu:.*/veth_mtu: \"1430\"/g" | kubectl --kubeconfig=./${CLUSTER_NAME}.kubeconfig apply -f -
+    curl https://docs.projectcalico.org/v3.23/manifests/calico.yaml | kubectl --kubeconfig=./${CLUSTER_NAME}.kubeconfig apply -f -
```
Comment (apricote) [inline docs/book/src/topics/external-cloud-provider.md:27]: Is there a reason why the lower `veth_mtu` is no longer required?

Comment (jichenjc) [reply docs/book/src/topics/external-cloud-provider.md:27]: curl https://docs.projectcalico.org/v3.23/manifests/calico.yaml 

I checked the yaml itself says 0 (by default) it's will auto detect it ..I think we should let CNI detect
and at least I can deploy a cluster with such setting ..

Comment (tobiasgiese) [reply docs/book/src/topics/external-cloud-provider.md:27]: In this case you're able to

```suggestion
    kubectl --kubeconfig=./${CLUSTER_NAME}.kubeconfig apply -f https://docs.projectcalico.org/v3.23/manifests/calico.yaml
```

Comment (apricote) [reply docs/book/src/topics/external-cloud-provider.md:27]: FWIW I just changed the veth_mtu on my test cluster to "0" and everything still works.

**Pull request merged**